CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -230,7 +223,35 @@ class CameraFragment : Fragment() {
      */
     override fun onConfigurationChanged(newConfig: Configuration) {
         super.onConfigurationChanged(newConfig)
+
+        // Redraw the camera UI controls
         updateCameraUi()
+
+        // Enable or disable switching between cameras
+        updateCameraSwitchButton()
+    }
+
+    /** Initialize CameraX, and prepare to bind the camera use cases  */
+    private fun setUpCamera() {
+        val cameraProviderFuture = ProcessCameraProvider.getInstance(requireContext())
+        cameraProviderFuture.addListener(Runnable {
+
+            // CameraProvider
+            cameraProvider = cameraProviderFuture.get()
+
+            // Select lensFacing depending on the available cameras
+            lensFacing = when {
+                hasBackCamera() -> CameraSelector.LENS_FACING_BACK
+                hasFrontCamera() -> CameraSelector.LENS_FACING_FRONT
+                else -> throw IllegalStateException("Back and front camera are unavailable")
+            }
+
+            // Enable or disable switching between cameras
+            updateCameraSwitchButton()
```

Comment:
Would it make more sense to move this inside of `updateCameraUi()`? If it doesn't make the method too big, consider inlining this too, to avoid creating too many methods.

Reply:
`updateCameraSwitchButton()` and `updateCameraUi()` are separate because when the Fragment is first created:
1. The UI controls are drawn (`updateCameraUi()`) (this disables the button to switch cameras)
2. CameraX is initialized (`cameraProviderFuture ...`)
3. The camera toggling button is updated (`updateCameraSwitchButton()`)

If we want to wait for CameraX to be initialized before drawing the UI controls, the 2 operations can be combined as you suggested. Camera setup can take some time, not too long that drawing the UI controls would wait for a whole lot, but it might be noticeable on some low-end devices. I'm okay with either approach, let me know which works best for you.

Reply:
I see what you mean. I agree that setting up the UI without waiting for camera initialization makes sense.